designcompose/src/main/java/com/android/designcompose/ComputePaths.kt

```diff
@@ -69,26 +69,78 @@ class ComputedPaths(
     var strokeCap: androidx.compose.ui.graphics.StrokeCap?,
 )
 
-class ComputedShadow(val fills: List<Path>, val shadowStyle: BoxShadow)
+internal class ComputedShadow(val fills: List<Path>, val shadowStyle: BoxShadow)
+
+/// Maintain a cache of computed paths, as some paths and strokes are quite expensive to
+/// compute (if we need to provide a stroke for a path), causing `computePaths` to show
+/// up prominently in profiles for Squoosh without a cache.
+internal class ComputedPathCache {
+    internal class Entry(
+        val computedPaths: ComputedPaths,
+        // We could be more selective than using the entire style, and styles are
+        // immutable so we could also simply do object identity comparison rather
+        // than a deep equality comparison.
+        val viewShape: ViewShape,
+        val style: ViewStyle,
+        val density: Float,
+        val frameSize: Size,
+        val overrideSize: Size?,
+        val customArcAngle: Boolean,
+        val variableState: VariableState,
+    )
+
+    private var cache: HashMap<Int, Entry> = HashMap()
+    private var nextGeneration: HashMap<Int, Entry> = HashMap()
+
+    fun get(layoutId: Int): Entry? {
+        return cache[layoutId]
+    }
+
+    fun put(layoutId: Int, data: Entry) {
+        nextGeneration[layoutId] = data
+    }
+
+    fun collect() {
```

Comment:
Can you add a comment explaining why you need to have a current, nextGeneration, and this `collect()` function?

Reply:
Yep, added some comments, hopefully that's better (I didn't change the method names, but I could do that, too).